### ISSUE TITLE
Configurable numeric keyboard

### ISSUE THREAD

**Issue by MisterY** [florisboard/florisboard#397]
description coming...

**Comment by MisterY** [comment]
Image 1, Regular keyboard

<img src="https://user-images.githubusercontent.com/462445/109140548-fcfbf780-775c-11eb-91cb-57aea8d94e6f.png" width="250px" />

Image 2, Numeric keyboard

<img src="https://user-images.githubusercontent.com/462445/109140564-04230580-775d-11eb-9a2d-2d162a5ca179.png" width="250px" />

Image 3, Special chars keyboard

<img src="https://user-images.githubusercontent.com/462445/109140835-5106dc00-775d-11eb-926d-2164873148c3.png" width="250px" />

Image 4, phone-numeric keyboard

<img src="https://user-images.githubusercontent.com/462445/109141496-1782a080-775e-11eb-8758-7b18db8cc272.png" width="250px" />

**Comment by patrickgold** [comment]
Thanks for your suggestion, will place this on my todo list!

**Comment by MisterY** [comment]
Thank you! The button should indicate which numeric keyboard will open. ("?123" or "1234")

**Comment by MisterY** [comment]
I must note that this pushes the special characters (image 3) one more tap away and requires some thought.

**Comment by zachdecook** [comment]
I've been using florisboard for programming python code for a week, and will note that this is my biggest gripe vs my custom ASK layout.
To type `['\n']` requires too many context switches between keyboard layers.

In my custom anysoftkeyboard layout, I can get the printable ASCII characters either though either keypress, longpress, or shift-keypress, never going to the numbers or symbols layer.
I have about 34 keys used here, but this could be done in florisboard with less by the following:

* shift affecting `'`, `,`, and `.` (turning them into `"`, `<`, and `.` respectively.
* multiple special characters per popup (long-press and drag on s to get backslash)
* shift affecting popups (semicolon could be shift+longpress b), shifting could change square brackets to curly braces
* remove redundant popups (`'` is a popup of x, but is also its own key
etc
* longpress spacebar for underscore or hyphen
etc

**Comment by zachdecook** [comment]
I used to use Swiftkey, and the one thing I liked about its numbers/symbols page was how the numbers were in a numpad-like grid. Switching to it offered a better experience than the standard layout offered, without being a phone keypad with fewer buttons.

**Comment by zachdecook** [comment]
A little more on-topic: a way this could be implemented without complicating settings:

* Pressing '?123' could bring you to a numeric keyboard
* long-pressing '?123' could bring you to the symbols keyboard
* swiping up on '?123' could bring you to the phone-numeric keyboard

**Comment by zachdecook** [comment]
Alternative(?) proposal: longpress-release on '?123' temporarily opens the symbols keyboard, and when you let go, it types the symbol under your finger, and switches back to the last layer.

This would be great for hitting those 12 missing keys quickly without a big context switch.

It would be easy to find the symbols (rather than nested long-press or shifted long-press), with the exception, perhaps of `<`, which would be hiding under your thumb.

This would be minimally disruptive to the current behavior and layouts of florisboard.

**Comment by centralscrutinizer21** [comment]
Numpad keyboard (image 4) should also have the corrispondent letters. Many phonebooks app allow you to search contacts by digiting on numpad and giving name results

**Comment by now0clock** [comment]
It needs letters on the numpad, like a phone's dialer. I've memorized numbers/words this way forever..
![Screenshot_20220316-080621](https://user-images.githubusercontent.com/59995403/158625557-800dc095-f2e0-4139-8d77-09a857ba2cd4.png)

**Comment by MisterY** [comment]
> It needs letters on the numpad, like a phone's dialer. I've memorized numbers/words this way forever..

Could you please clarify why does it "need" to have the letters? What do they contribute to the functionality of the numeric keyboard? How do they make it better?
Would the additional clutter make it harder to read?

**Comment by now0clock** [comment]
> > It needs letters on the numpad, like a phone's dialer. I've memorized numbers/words this way forever..
> 
> Could you please clarify why does it "need" to have the letters? What do they contribute to the functionality of the numeric keyboard? How do they make it better? Would the additional clutter make it harder to read?

For me anyways, it needs the letters since I have always looked at a numpad that way, and use the reference for typing pin codes, etc. (a sequence of numbers that is also a word/phrase.)

**Comment by MisterY** [comment]
I agree and like the idea, for a dial-pad. But I'm not sure the Image 4 is a dialpad keyboard. It looks more like a numeric pad type, with additional keys that do not exist on the phone pad. Anyways, it is a nice idea. Please create a new issue so that related discussion and arguments can be kept related.
